barrels/LogMonkey/LogTests.mc

```diff
@@ -0,0 +1,246 @@
+using Toybox.Test as Test;
+using Toybox.Lang as Lang;
+using LogMonkey as Log;
+
+module LogMonkey {
+
+    // Wrap the unit tests in a module annotated with :test so it will get compiled
+    // out for non unit test builds.
+    (:test)
+    module Test {
+
+        //! This class mocks the Toybox.System.println() interface but instead
+        //! of printing a value saves it to a string value which can be retrieved
+        //! and examed to verify log results are correct.
+        class MockSys {
```

Comment:
This class should get an `initialize()` method to avoid compiler warnings for users that enable them.

Reply:
Unless we use the initialize function to set mStreamContents to an empty string as suggested below - which is ultimately what I did - I don't think this is necessary. This class isn't overriding another class so there's not a parent initialize to call and the compiler will insert an empty initialize function if you don't include one so any class that would subclass this one would be able to call the initialize symbol safely.

Reply:
You are right, the warning would only occur if there was a base.